### PR TITLE
Fix bug 1686690: simplify 'otherlocales' response payload

### PR DESCRIPTION
**Pull request by julen**
* Unifies the initial state values to match what machinery uses
* Flattens the response payload, simplifies frontend code

### CODE REVIEW

File: pontoon/base/views.py [hunk 1]
```diff
@@ -334,6 +334,7 @@ def _serialize_translation_values(query):
                 "script": translation["locale__script"],
             },
             "translation": translation["string"],
+            "is_preferred": translation["locale__code"] in preferred_values,
```
Comment (julen) [inline pontoon/base/views.py:337]: We could leave this out entirely if the value is `False` and hence reduce the response payload.

**Review by mathjazz — COMMENTED** [review]
Nice job, this is much cleaner.

It seems to be working very well. I only caught two `eslint` warnings (note that GH Actions are down ATM):
```
./src/modules/otherlocales/components/OtherLocales.js
  Line 64:60:  Expected to return a value at the end of arrow function  array-callback-return
  Line 72:60:  Expected to return a value at the end of arrow function  array-callback-return
```

**Review by mathjazz — APPROVED** [review]
Thanks for the update! This is good to go.

P.S.: I saw the eslint errors in the output of `make run`.

**Comment by mathjazz** [comment]
@julen Do you want to squash commits and push them to your branch before I merge?

**Comment by julen** [comment]
There you go!

**Pull request merged**